src/common/modules/data/Tips.js

```diff
@@ -0,0 +1,136 @@
+ * with a chance of 20%, see {@link RandomTips.GLOBAL_RANDOMIZE}.
+ * @property {string} text The text to actually show. It is passed to the
+ * {@link MessageHandler}, so you can (& should) use a translatable string here.
+ * @property {Object} [actionButton] adds an action button to the message, is
+ * passed as-is to the {@link MessageHandler} module.
+ * @property {string} actionButton.text the text to use for the button
+ * @property {string|function} actionButton.action the link or function to
+ * execute on click
+ * @property {callbackShowTip} [showTip] evaluate via a custom function
+ * whether to show or not to show the tip
+ */
+
+/**
+ * Custom function to check whether a tip shall be shown or not.
+ *
+ * **Attention:** This is evaluated before all other constraints were evaluated.
+ * Be careful before returning `true`, you may actually want to return `null`.
+ *
+ * * If you return `true`, it means you **force**(!) the tip to be shown. This
+ *   overrides the usual conditions and shows the tip immediately.
+ * * If you return `false`, it means you disallow the tip to be shown.
+ * * If `null` is returned, the usual (other) conditions that determinate whether
+ *   to show the tip are evaluated.
+ * If you throw something, this cancels the whole tip selection process.
+ *
+ * Note that you can actually _modify_ the values of the first two parameters, i.e.
+ * `tipSpec` and `thisTipConfig`. If you modify the config, it will be saved
+ * automatically.
+ * The difference between `tipSpec` and `tipSpecOrig` is, that in `tipSpec` the
+ * default values of the tips have been added/modified, if needed and that
+ * `tipSpecOrig` is the original as you have specified it in {@link TipObject}.
+ *
+ * For stylistic reasons, it is recommend to not hardcode the function into the
+ * this file, but use a separate file/module.
+ *
+ * @public
+ * @callback callbackShowTip
+ * @param  {TipObject} tipSpec the TipObject, merged with default values
+ * @param  {module:RandomTips~TipConfigObject} thisTipConfig the settings of the tip
+ * @param  {TipObject} tipSpecOrig the original, unmodified (frozen) TipObject
+ *                                  of the tip
+ * @param  {module:RandomTips~ModuleConfig} moduleConfig the whole config of this module
+ * @return {boolean|null}
+ */
+
+/**
+ * An array of all tips.
+ *
+ * @private
+ * @const
+ * @type {Array.<TipObject>}
+ */
+const tipArray = [
+    {
+        id: "likeAddon",
+        requiredShowCount: 3,
+        requireDismiss: 1,
+        maximumDismiss: 2,
+        requiredTriggers: 10,
+        showInContext: {
+            "popup": 1
+        },
+        randomizeDisplay: false,
+        text: "tipYouLikeAddon",
+        actionButton: {
+            text: "tipYouLikeAddonButton",
+            action: "https://addons.mozilla.org/firefox/addon/unicodify-text-transformer/reviews/?utm_source=unicodify-addon&utm_medium=addon&utm_content=unicodify-addon-tips-tipYouLikeAddon"
```

Comment:
Note that this and the below link will need to be updated to support Thunderbird (ATN) and the Chrome Web Store at some point.

Reply:
Indeed, as that needs to be done in RandomTips. So I've created an issue: https://github.com/TinyWebEx/RandomTips/issues/6

Reply:
The links are all static, so I believe it could be done directly in this file:
```js
action: IS_CHROME ? "https://chrome.google.com/webstore/detail/unicodify-text-transformer/..." : IS_THUNDERBIRD ? "https://addons.thunderbird.net/thunderbird/addon/unicodify-text-transformer/reviews/" : "https://addons.mozilla.org/firefox/addon/unicodify-text-transformer/reviews/"
```
It is not very pretty, but it should work.

Reply:
oh, yeah, it could, but I guess function would still be better.

Reply:
@tdulcet Found an existing solution. Actually, in RandomTips this is passed to the Message module, so it is harder to adjust if we want a function there. (Okay, well… would need to adjust it in Messages)

Anyway, I found a solution using `showTip` that also works asynchronously, I think… I've committed it, but not yet tested it (in all "borwsers"). Do you want?

Reply:
Sure, no problem. The `getBrowserUrl` function fails in Chrome for the same reason as the first screenshot in https://github.com/TinyWebEx/AutomaticSettings/issues/12#issuecomment-812993836. How about something like this:
```js
async function getBrowserUrl(switchBrowser) {
    if (browser.runtime.getBrowserInfo) {
        const browserInfo = await browser.runtime.getBrowserInfo();

        if (browserInfo.name === "Thunderbird") {
            return switchBrowser.thunderbird;
        } else {
            return switchBrowser.firefox;
        }
    } else {
        return switchBrowser.chrome;
    }
}
```

Reply:
Thanks, changed.

Reply:
No problem. Note that this function will probably also need to be used for the Awesome Emoji Picker link on the options page.